### ISSUE TITLE
Build error throws when running dotnet pack with multiple target framework

### ISSUE THREAD

**Issue by Karthi2414** [dotnet/aspnetcore#30427]
### Bug Description:

After upgrading the .NET 6.0 preview 1 SDK, it throws build error when running dotnet pack command line. The problem is occur when the project file have multiple target frameworks `<TargetFrameworks>netstandard2.0;net5.0</TargetFrameworks>`. If I change it to single `<TargetFramewok>` then it works fine.
 

### Exceptions (if any)

> C:\Program Files\dotnet\sdk\6.0.100-preview.1.21103.13\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.ScopedCss.targets(134,3): **error MSB4044: The "ComputeCssScope" task was not given a value for the required parameter "TargetName".** [D:\Feb\NET6\RazorClassLibrary2\RazorClassLibrary2.csproj]

### To Reproduce
* Download the Razor sample.
* Goto command prompt and then run the command `dotnet pack -c Release`

### Sample:

[RazorClassLibrary3.zip](https://github.com/dotnet/aspnetcore/files/6035384/RazorClassLibrary3.zip)

### Further technical details
Blazor RazorClass Library 6.0 sample
Latest SDK .NET 6.0 Preview 1
VS 2019 preview

**Comment by pranavkm** [comment]
Thanks for the issue report. @captainsafia this looks like a regression as part of our change to use a source generator. Static Web Assets appears to wire up to `RazorComponentGenerate` which is no longer invoked with source generators. I'm able to reproduce this with the latest SDK build.

**Comment by captainsafia** [comment]
I've been able to reproduce this on preview1 and preview2 so I don't think it's strictly related to source generators.

However, we'll likely have to make different modifications to the ordering for the source generator/non-source generator scenarios.

**Comment by icsilambarasan** [comment]
As per the above comment, this issue persists in v6.0.0-preview.2 released yesterday. @pranavkm / @captainsafia Can we have any tentative timeline to expect the fix or having any workaround in the meantime?

**Comment by skclusive** [comment]
any update on the issue?. still reproducible in v6.0.0-preview3. not able to multi target component libraries due to this issue. hope the issue will be fixed in next preview. thanks

**Comment by captainsafia** [comment]
@skclusive None yet. We're tracking it under next sprint planning. If you're affected by this, you can let us know by giving the issue a thumbs up.

**Comment by skclusive** [comment]
@captainsafia thanks. hope it will be considered for next sprint. would like to convey that this is more than a blazor bug. it is regression of basic expectation of dotnet cli. (dotnet pack to multi target C# library)

even tried the following from documentation with no luck. https://docs.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?view=aspnetcore-5.0 


    <DisableScopedCssBundling>true</DisableScopedCssBundling>

**Comment by kmuthukumar23** [comment]
@pranavkm / @captainsafia - Any updates for this? Can you please share any expected timeline for the fix?

**Comment by captainsafia** [comment]
@kmuthukumar23 We haven't had the chance to look at this yet. If this issue impacts you, the best thing to do is to give the original issue a thumbs up. This helps us out when we do triage to identify which issues need to be fixed.

I've gotten as far as reproducing this and starting to tinker with the binlog to see what's up but haven't landed on a fix.

**Comment by robertmclaws** [comment]
Still running into this issue with Preview 7.

**Comment by robertmclaws** [comment]
To help you folks in troubleshooting, it happens when the `<TargetFrameworks>` tag is present, regardless of whether you are *actually* multitargeting a project or not. Even if you have one value in there, it still happens.

**Comment by robertmclaws** [comment]
I believe that this code might be the problem: https://github.com/dotnet/sdk/blob/be0e8ec2065fabc89ecb967fcd0b3aeb43d1b9d7/src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.ScopedCss.targets#L53-L83

The referenced property groups almost cancel each other out. Since all MSBuild true/false values are actually strings, I had to set DisableScopedCssBundling to something else in order to ACTUALLY disable it.

**WORKAROUND**: I was able to get multitargeting (for .NET 6.0 and .NET 5.0) to work in Preview 7 by adding the following to the project file:
```xml
  <PropertyGroup>
    <DisableScopedCssBundling>none</DisableScopedCssBundling>
    <EnableDefaultScopedCssItems>false</EnableDefaultScopedCssItems>
    <ScopedCssEnabled>false</ScopedCssEnabled>
  </PropertyGroup>

  <Target Name="PrepareProjectReferences">
  </Target>
```

The blank Target was added because `Microsoft.NET.Sdk.Razor.StaticWebAssets.5_0.targets` requires the target to be specified.

BTW, did you notice something in the PropertyGroup? Each property enables some part of CSS scoping, and each is named inconsistently with the others. These are the kinds of situations where I wish the dev team would pick a naming convention with public-facing MSBuild properties, and then stick to it.

Hopefully this helps you folks track it down in time for RC1.

**Comment by kmuthukumar23** [comment]
@robertmclaws - Thanks for your workaround.

I have tried with net5.0 and .net6.0 in the attached demo project with your workaround changes in .csproj with .net6 preview 7 sdk. And the run the 'dotnet build' and 'dotnet pack -c Release' in cli. dotnet build has success. but i have faced the errors with static assets when run the dotnet pack command.

`C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.5_0.targets(436,5): error : Static web asset 'C:\Users\User\Downloads\RazorClassLibrary3\wwwroot\exampleJsInterop.js' has invalid source type ''. [C:\Users\User\Downloads\RazorClassLibrary3\RazorClassLibrary3.csproj]`

Same error thrown when adding CSS files as static assets.
`C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.5_0.targets(436,5): error : Static web asset 'C:\Users\User\Downloads\RazorClassLibrary3\wwwroot\test.css' has invalid source type ''. [C:\Users\User\Downloads\RazorClassLibrary3\RazorClassLibrary3.csproj]`

Please find the attached project below.

[RazorClassLibrary3.zip](https://github.com/dotnet/aspnetcore/files/7122468/RazorClassLibrary3.zip)


This is my SDK lists.
![image](https://user-images.githubusercontent.com/31958591/132366843-da67279e-a3af-4d38-9316-03bf08e7d4d1.png)

How can resolve this errors? Can anyone help on this?

Thanks in advance....

**Comment by javiercn** [comment]
This was fixed in https://github.com/dotnet/sdk/pull/20609 and will be available on 6.0-rc2
